src/stream/src/executor/top_n/top_n_cache.rs

```diff
@@ -309,38 +310,46 @@ impl TopNCacheTrait for TopNCache<false> {
 impl TopNCacheTrait for TopNCache<true> {
     fn insert(
         &mut self,
-        ordered_pk_row: OrderedRow,
+        cache_key: CacheKey,
         row: Row,
         res_ops: &mut Vec<Op>,
         res_rows: &mut Vec<CompactedRow>,
     ) {
         assert!(self.low.is_empty());
 
-        let elem_to_compare_with_middle = (ordered_pk_row, row);
+        let elem_to_compare_with_middle = (cache_key, row);
 
         if !self.is_middle_cache_full() {
             self.middle.insert(
-                elem_to_compare_with_middle.0,
+                elem_to_compare_with_middle.0.clone(),
                 (&elem_to_compare_with_middle.1).into(),
             );
             res_ops.push(Op::Insert);
             res_rows.push((&elem_to_compare_with_middle.1).into());
             return;
         }
 
-        let sort_key = elem_to_compare_with_middle.0.prefix(self.order_by_len);
+        let sort_key = &elem_to_compare_with_middle.0 .0;
         let middle_last = self.middle.last_key_value().unwrap();
-        let middle_last_order_by = middle_last.0.prefix(self.order_by_len);
+        let middle_last_order_by = &middle_last.0 .0.clone();
 
-        match sort_key.cmp(&middle_last_order_by) {
+        match sort_key.cmp(middle_last_order_by) {
             Ordering::Less => {
                 // The row is in middle.
-                let num_ties = self.middle.range(middle_last_order_by.clone()..).count();
```

Comment:
Why to change this?

Reply:
Since `CacheKey = (Vec<u8>, Vec<u8>)`, it's hard to count prefix range...

Reply:
`let num_ties = self.middle.range((middle_last_order_by.clone(),vec![])..).count();`

Reply:
Oh amazing, Thanks for your suggestion!❤️